src/APIBasedAuth.ts

```diff
@@ -88,30 +86,111 @@ class APIBasedAuth {
         APIBasedAuth.PasswordlessAuthData
       >('/passwordless-auth', {
         appsBaseUri,
-        clientId: this.clientId,
+        clientId: this.clientOptions.clientId,
         loginAppBasePath,
         username
       })
-      .then(({ data }) => {
-        this.session = {
-          session: data.session,
-          username
-        };
+      .then(async ({ data }) => {
+        await this._store('session', { session: data.session, username });
         return data;
       })
       .catch(formatAxiosError);
   }
 
-  public logout() {
-    this.session = undefined;
+  public async logout() {
+    await this._removeFromStorage('session');
+    await this._removeFromStorage('token');
+  }
+
+  private _getStorageKeyFromType<T extends APIBasedAuth.StorageTypeName>(
+    type: T
+  ) {
+    if (type === 'session') {
+      return ['session', 'username'];
+    }
+
+    if (type === 'token') {
+      return ['accessToken', 'idToken', 'refreshToken'];
+    }
+  }
+
+  private async _getFromStorage<T extends APIBasedAuth.StorageTypeName>(
+    type: T
+  ): Promise<APIBasedAuth.StorageType<T>> | undefined {
+    if (type === 'session' && this.session) {
+      return this.session as APIBasedAuth.StorageType<T>;
+    }
+
+    if (type === 'token' && this.tokens) {
+      return this.tokens as APIBasedAuth.StorageType<T>;
+    }
```

Comment:
I don't see any call-sites where we do `_getFromStorage('token')` -- am I missing something?

Reply:
you're correct but the functionality exists for the potential this class could be expanded to potentially need it

Reply:
woah, why 100? it's already covering 100% for functions. 95% for statements/lines. ~88% for branches

Reply:
@smolster alright removed the unused stuff for now and we can add to this in the future if/when needed